HeterogeneousCore/SonicTriton/interface/TritonData.h

```diff
@@ -91,7 +124,11 @@ class TritonData {
   std::string dname_;
   inference::DataType dtype_;
   int64_t byteSize_;
-  std::any holder_;
+  size_t sizeShape_;
+  size_t byteSizePerBatch_;
+  size_t totalByteSize_;
+  mutable std::shared_ptr<void> holder_;
```

Comment:
In `TritonOutputData`, when GPU shared memory is used, the output values have to be copied back from the GPU. The values are then stored in `holder_` to extend their lifetime. The call to copy back from the GPU is issued in `fromServer()`, but the SONIC interface specifies read-only access to the output object in `produce()`. I view this as a correct use of `mutable` because it's something happening "behind the scenes" and the "user" (module developer) should not otherwise be able to change the output object itself.

Reply:
I'm unable to find a place in this PR (or in `master`) where `holder_` is used by a non-const member function.

In principle "copy from" is a const operation on the "from" object.

Reply:
Actually, you're right. It was needed before I refactored the shared memory operations into the separate `TritonMemResource` class.

`fromServer() const` calls `copy()`: https://github.com/cms-sw/cmssw/blob/b5e974956bfb809e6aeb383a76ffb0e8f265ceb1/HeterogeneousCore/SonicTriton/src/TritonData.cc#L226
which modifies `holder_`: https://github.com/cms-sw/cmssw/blob/b5e974956bfb809e6aeb383a76ffb0e8f265ceb1/HeterogeneousCore/SonicTriton/src/TritonMemResource.cc#L159

But since `TritonMemResource` has a non-const pointer to `TritonData` and `copy()` itself isn't `const`, this is allowed without `mutable`.